dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleAction.java

```diff
@@ -134,6 +137,15 @@ public class ProgramRuleAction extends BaseIdentifiableObject implements Metadat
    *   <li>sendmessage
    * </ul>
    */
+  @JsonIgnore private ProgramNotificationTemplate notificationTemplate;
+
+  /**
+   * The program notification template uid that will be triggered by the rule action. Used for:
+   *
+   * <ul>
+   *   <li>sendmessage
+   * </ul>
+   */
   private String templateUid;
```

Comment:
Could you use `@JsonSetter` & `@JsonGetter` on the relevant methods and get rid of this property completely? It doesn't look like it serves a purpose & muddies the Object (2 properties that do the same thing but this isn't used anywhere that I can see, unless I'm missing something).
I see the `templateuid` column is being removed from the DB also which is good, so it would make sense to remove it here also I think.
Jackson should be flexible enough to keep the same API, while using the new `notificationTemplate` property.